### PR TITLE
Backport kvm ioctls fix

### PR DESCRIPTION
**Pull request by roypat**
## Changes

Backport those changes of https://github.com/firecracker-microvm/firecracker/pull/3280 to firecracker 1.1 required to fix snapshots not saving the upper 64 bit of the V0-V31 FL/SIMD registers on ARM

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license.

## PR Checklist

- [ ] All commits in this PR are signed (`git commit -s`).
- [ ] If a specific issue led to this PR, this PR closes the issue.
- [ ] The description of changes is clear and encompassing.
- [ ] Any required documentation changes (code and docs) are included in this PR.
- [ ] New `unsafe` code is documented.
- [ ] API changes follow the [Runbook for Firecracker API changes][2].
- [ ] User-facing changes are mentioned in `CHANGELOG.md`.
- [ ] All added/changed functionality is tested.

---

- [ ] This functionality can be added in [`rust-vmm`][1].

[1]: https://github.com/rust-vmm
[2]: https://github.com/firecracker-microvm/firecracker/blob/main/docs/api-change-runbook.md

### CODE REVIEW

File: src/arch/src/aarch64/regs.rs [hunk 1]
```diff
@@ -151,12 +172,11 @@ arm64_sys_reg!(MIDR_EL1, 3, 0, 0, 0, 0);
 ///
 /// # Arguments
 ///
-/// * `state` - Array slice of kvm_one_reg structures, representing
-///             the registers of a VCPU state.
-pub fn get_manufacturer_id_from_state(state: &[kvm_one_reg]) -> Result<u32> {
+/// * `state` - Array slice of kvm_one_reg structures, representing the registers of a VCPU state.
```
Comment (luminitavoicu) [inline src/arch/src/aarch64/regs.rs:175]: Nit: comment is out of date

File: tests/integration_tests/functional/test_mmds.py [hunk 2]
```diff
@@ -825,7 +825,7 @@ def test_mmds_snapshot(bin_cloner_path,  version):
     # artifact.
     firecracker_artifacts = artifacts.firecrackers(
         # v1.0.0 breaks snapshot compatibility with older versions.
```
Comment (luminitavoicu) [inline tests/integration_tests/functional/test_mmds.py:827]: Nit: outdated comment

File: tests/integration_tests/functional/test_snapshot_advanced.py [hunk 3]
```diff
@@ -90,7 +91,7 @@ def test_restore_old_version(bin_cloner_path):
     # artifact.
     firecracker_artifacts = artifacts.firecrackers(
         # v1.0.0 breaks snapshot compatibility with older versions.
```
Comment (luminitavoicu) [inline tests/integration_tests/functional/test_snapshot_advanced.py:93]: Nit: same as above, outdated comment

File: tests/integration_tests/build/test_coverage.py [hunk 4]
```diff
@@ -31,7 +31,7 @@
 if utils.is_io_uring_supported():
     COVERAGE_DICT = {"Intel": 83.83, "AMD": 83.15, "ARM": 83.01}
 else:
-    COVERAGE_DICT = {"Intel": 80.94, "AMD": 80.26, "ARM": 80.17}
+    COVERAGE_DICT = {"Intel": 80.94, "AMD": 80.26, "ARM": 80.12}
```
Comment (luminitavoicu) [inline tests/integration_tests/build/test_coverage.py:34]: Is this fetched from the failure message or back-ported from #3280?

File: tests/framework/artifacts.py [hunk 5]
```diff
@@ -304,7 +304,7 @@ class ArtifactCollection:
     PLATFORM = platform.machine()
 
     # S3 bucket structure.
-    ARTIFACTS_ROOT = 'ci-artifacts'
+    ARTIFACTS_ROOT = 'ci-artifacts-dev'
```
Comment (luminitavoicu) [inline tests/framework/artifacts.py:307]: Leaving this comment as a reminder to revert before merging.

**Review by luminitavoicu — COMMENTED** [review]
Looking good, just a few nits! Not approving yet because we still need to undo a commit.

It would be useful to specify in the PR description why we are not backporting also the updates to vm-superio and event-manager crates.

File: tests/integration_tests/build/test_coverage.py (hunk 4 above)
Comment (roypat) [reply tests/integration_tests/build/test_coverage.py:34]: These are taken from the failure messages

File: tests/integration_tests/functional/test_mmds.py (hunk 2 above)
Comment (roypat) [reply tests/integration_tests/functional/test_mmds.py:827]: ooo good catch. I also removed some code that specifically referred to 1.0.0

File: src/arch/src/aarch64/regs.rs (hunk 1 above)
Comment (roypat) [reply src/arch/src/aarch64/regs.rs:175]: this should probably also get fixed on main - I'll make a note to do a follow up PR with some cleanup once I have the time

**Review by luminitavoicu — APPROVED** [review]
Good catch removing older mmds logic!

**Review by aghecenco — APPROVED** [review]

**Pull request merged**